conda_smithy/lint_recipe.py

```diff
@@ -536,17 +539,40 @@ def _format_validation_msg(error: "jsonschema.ValidationError"):
     )
 
 
-def main(recipe_dir, conda_forge=False, return_hints=False):
+def main(
+    recipe_dir, conda_forge=False, return_hints=False, feedstock_dir=None
+):
     recipe_dir = os.path.abspath(recipe_dir)
-    recipe_meta = os.path.join(recipe_dir, "meta.yaml")
-    if not os.path.exists(recipe_dir):
-        raise IOError("Feedstock has no recipe/meta.yaml.")
+    build_tool = CONDA_BUILD_TOOL
+    if feedstock_dir:
```

Comment:
Should we default feedstock_dir to "{recipe_dir}/.." if it is not given?

Reply:
I would prefer to not doing this so we can enable linting of recipe.yaml by explicitly passing feedstock-dir. Also, making it a default would require additional changes to the test suite

Reply:
The defaults can be overriden at the command line, right? I don't follow.

Reply:
It can be, but we have the same behavior in conda-forge-ci-setup: `https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/316/files#diff-2d7dc8fce520979cf844640e36ee6becb8e393f62efc53fea141ac8629f2b6eaR362`

I would prefer them to be explicit until we fully integrate the `rattler-build` so we could gradually enabling them